### DISCUSSION TITLE
Correct usage with latest brew

### DISCUSSION THREAD

**Discussion by infa-jowoods** [Homebrew/discussions#2058] [Everyday usage] [answered]
I've had brew installed on my M1/Apple Silicon Mac for a few months.

I recently installed in on another Mac and noticed a difference during install...

On my first Mac, all I have in `.zshrc` is: 

`"export PATH=/opt/homebrew/bin:$PATH"`
`"export PATH=/opt/homebrew/sbin:$PATH"`

however, I noticed on the new Mac install of home-brew it instead instructed me to add this:

`eval "$(/opt/homebrew/bin/brew shellenv)"`

I do have this file, should I ensure this is part of `.zshrc` too? 
Things seem to work fine so far, not sure when this was added and wanted to check if it's needed?

It seems to me that the latter sets some environment vars and indeed the paths I was manually setting anyway... strange

**Comment by gromgit** [comment, marked as answer]
You should replace _all_ your manual Homebrew `PATH` settings with a single `eval ... shellenv`, as the latter sets _all_ relevant environment variables (e.g. `MANPATH` and `INFOPATH`), and is smart enough to _not_ set stuff if it's already set (so you don't end up with overlong paths full of duplicates).

**Reply by infa-jowoods** [reply]
Thank you!
